components/feature/search/src/main/java/mozilla/components/feature/search/region/RegionManager.kt

```diff
@@ -0,0 +1,123 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package mozilla.components.feature.search.region
+
+import android.content.Context
+import android.content.SharedPreferences
+import androidx.annotation.VisibleForTesting
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.delay
+import kotlinx.coroutines.withContext
+import mozilla.components.browser.state.search.RegionState
+import mozilla.components.service.location.LocationService
+
+// The amount of time (in seconds) we need to be in a new
+// location before we update the home region.
+// Currently set to 2 weeks.
+// https://searchfox.org/mozilla-central/rev/89d33e1c3b0a57a9377b4815c2f4b58d933b7c32/toolkit/modules/Region.jsm#82-85
+private const val UPDATE_INTERVAL = 60 * 60 * 24 * 14
```

Comment:
nit: UPDATE_INTERVAL_IN_SECONDS, but see comment below. I think this was meant to be in MS as well and missing a `*1000`.